File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        
        HashMap<Integer, Integer> hm = new HashMap<>();
        int max =0;
        int maxvalue =0;
        for(int i=0;i<nums.length;i++){
            // if(hm.containsKey(nums[i])){

            // }
            hm.put(nums[i], hm.getOrDefault(nums[i],0)+1);

        }

        for(Map.Entry<Integer,Integer> kv : hm.entrySet()){
            if(max<kv.getValue()){
                max = kv.getValue();
                maxvalue = kv.getKey();
            }
        
    }
    return maxvalue;
    }
}

File: Data Structures & Algorithms/find-pivot-index/submission-0.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        pre = []
        n = len(nums)
        pre.append(nums[0])
        for i in range(1, n):
            pre.append(pre[i-1]+nums[i])
        for i in range(n):
            l = 0
            r = 0
            if i == 0:
                r = pre[n-1] - pre[i]
            else:
                l = pre[i-1]
                r = pre[n-1] - pre[i]
            if l == r:
                return i
                
        return -1